programs/marginfi/src/instructions/marginfi_account/liquidate.rs

```diff
@@ -249,39 +249,14 @@ pub fn lending_account_liquidate<'info>(
                 &mut liquidator_marginfi_account.lending_account,
             )?;
 
-            // TODO: in edge cases, the liquidator starts/ends with ASSETS, following the logic
-            // below, and this value (which will ultimately be emitted in the event) is useless.
+            // TODO: in edge cases, the liquidator starts/ends with ASSETS,
+            // and this value (which will ultimately be emitted in the event) is useless.
             let pre_balance: I80F48 = bank_account
                 .bank
                 .get_liability_amount(bank_account.balance.liability_shares.into())?;
 
-            let asset_balance: I80F48 = bank_account
-                .bank
-                .get_asset_amount(bank_account.balance.asset_shares.into())?;
-
-            // Complex cases: the liquidator has an asset balance, which will be REDUCED
-            if asset_balance > I80F48::ZERO {
-                // a) liquidator has assets ≥ needed → just withdraw that amount
-                if asset_balance >= liab_amount_liquidator {
-                    bank_account.withdraw(liab_amount_liquidator, true)?;
-                } else {
-                    // b) has some assets < needed → wipe them all, then borrow the rest
-                    bank_account.withdraw_all(true)?;
-                    let remaining = liab_amount_liquidator
-                        .checked_sub(asset_balance)
-                        .ok_or(MarginfiError::MathError)?;
-                    // Note: the slot is now deactivated, we must recreate it
-                    bank_account = BankAccountWrapper::find_or_create(
-                        &ctx.accounts.liab_bank.key(),
-                        &mut liab_bank,
-                        &mut liquidator_marginfi_account.lending_account,
-                    )?;
-                    bank_account.borrow_ignore_borrow_cap(remaining)?;
-                }
-            } else {
-                // Common case: liquidator has debt, or this is a new borrow
-                bank_account.borrow_ignore_borrow_cap(liab_amount_liquidator)?;
-            }
+            // Liquidator will withdraw the collateral (if any) and then borrow the remainder (if any). Ignores the utilization ratio check.
+            bank_account.withdraw_ignore_borrow_cap(liab_amount_liquidator)?;
```

Comment:
This is what it already had originally, and I removed it in favor of the "full" logic that doesn't involve flipping, because having this flip logic handled internally is seriously hard to follow (for me). Also I think to fix the event emit later, we will need the full logic: it should emit differently for a withdraw vs a withdraw-borrow that flips.

Well if you prefer the flipping version (it's certainly shorter!) then no objection on my part.